test/instrumentations/amqplib/amqplib.test.ts

```diff
@@ -0,0 +1,145 @@
+import { SpanKind } from '@opentelemetry/api';
+import * as fs from 'fs';
+import 'jest-expect-message';
+import 'jest-json';
+import { join } from 'path';
+import { GenericContainer, StartedTestContainer, Wait } from 'testcontainers';
+import { itTest } from '../../integration/setup';
+import { getSpanByName, readSpanDump } from '../../utils/spans';
+import { TestApp } from '../../utils/test-apps';
+import { installPackage, reinstallPackages, uninstallPackage } from '../../utils/test-setup';
+import { sleep } from '../../utils/time';
+import { versionsToTest } from '../../utils/versions';
+import {
+  filterAmqplibSpans,
+  getExpectedResourceAttributes,
+  getExpectedSpan,
+} from './amqplibTestUtils';
+
+const DEFAULT_RABBITMQ_PORT = 5672;
+const INSTRUMENTATION_NAME = `amqplib`;
+const SPANS_DIR = join(__dirname, 'spans');
+const STARTUP_TIMEOUT = 30_000;
+const TEST_APP_DIR = join(__dirname, 'app');
+const TEST_TIMEOUT = 600_000;
+
+const startRabbitMqContainer = async () => {
+  return await new GenericContainer('rabbitmq:latest')
+    .withExposedPorts(DEFAULT_RABBITMQ_PORT)
+    .withWaitStrategy(Wait.forLogMessage('Server startup complete'))
+    .withStartupTimeout(STARTUP_TIMEOUT)
+    .start();
+};
+
+describe.each(versionsToTest(INSTRUMENTATION_NAME, INSTRUMENTATION_NAME))(
+  `Instrumentation tests for the ${INSTRUMENTATION_NAME} package`,
+  function (versionToTest) {
+    let testApp: TestApp;
+    let rabbitmqContainer: StartedTestContainer;
+
+    beforeAll(async function () {
+      reinstallPackages(TEST_APP_DIR);
+      fs.mkdirSync(SPANS_DIR, { recursive: true });
+      installPackage(TEST_APP_DIR, INSTRUMENTATION_NAME, versionToTest);
+
+      /*
+       * Warm up container infra, download images, etc.
+       * This prevents spurious failures of early tests.
+       */
+      try {
+        rabbitmqContainer = await startRabbitMqContainer();
+      } finally {
+        if (rabbitmqContainer) {
+          rabbitmqContainer.stop();
+        }
+      }
+    }, STARTUP_TIMEOUT /* Long timeout, this might have to pull Docker images */);
```

Comment:
maybe it's worth adding a print before it?
So when you have slow network you'll know that this is the reason